File: app/src/main/java/com/my/televip/obfuscate/ObfuscationManager.java
package com.my.televip.obfuscate;

import com.my.televip.Clients.ClientManager;
import com.my.televip.obfuscate.struct.ClientObfuscationData;
import com.my.televip.utils.JsonAssetReader;

import java.util.EnumMap;
import java.util.Map;

public class ObfuscationManager {

    private static final Map<ClientManager.Client, ClientObfuscationData> cache =
            new EnumMap<>(ClientManager.Client.class);

    private static volatile ClientObfuscationData currentClientData;
    private static volatile boolean attemptedLoad = false;

    public static ClientObfuscationData current() {
        if (currentClientData != null) return currentClientData;
        if (attemptedLoad) return null;

        synchronized (ObfuscationManager.class) {
            if (currentClientData != null) return currentClientData;
            attemptedLoad = true;

            ClientManager.Client type = ClientManager.getCurrent();
            if (type == null) return null;

            currentClientData = load(type);
            return currentClientData;
        }
    }

    private static ClientObfuscationData load(ClientManager.Client type) {
        if (cache.containsKey(type)) return cache.get(type);

        ClientObfuscationData data = null;
        String clientPath = "assets/clients/" + type.name() + ".json";
        String clientJson = JsonAssetReader.readRaw(clientPath, false);

        String clientAliasPath = "assets/clients/Alias/" + type.name() + ".json";
        String clientAliasJson = JsonAssetReader.readRaw(clientAliasPath, false);

        if (clientJson != null && clientAliasJson != null)
            data = ClientObfuscationData.parse(clientJson, clientAliasJson);

        if (data != null)
            cache.put(type, data);
        return data;
    }

}

File: app/src/main/java/com/my/televip/obfuscate/ObfuscationManager.kt
package com.my.televip.obfuscate

import com.my.televip.clients.ClientManager
import com.my.televip.obfuscate.struct.ClientObfuscationData
import com.my.televip.utils.JsonAssetReader
import java.util.EnumMap

object ObfuscationManager {

    private val cache: MutableMap<ClientManager.Client, ClientObfuscationData> =
        EnumMap(ClientManager.Client::class.java)

    @Volatile
    private var currentClientData: ClientObfuscationData? = null

    @Volatile
    private var attemptedLoad = false

    @JvmStatic
    fun current(): ClientObfuscationData? {
        currentClientData?.let { return it }
        if (attemptedLoad) return null

        synchronized(this) {
            currentClientData?.let { return it }
            attemptedLoad = true

            val type = ClientManager.getCurrent() ?: return null
            currentClientData = load(type)
            return currentClientData
        }
    }

    private fun load(type: ClientManager.Client): ClientObfuscationData? {
        if (cache.containsKey(type)) return cache[type]

        var data: ClientObfuscationData? = null
        val clientPath = "assets/clients/${type.name}.json"
        val clientJson = JsonAssetReader.readRaw(clientPath, false)

        val clientAliasPath = "assets/clients/Alias/${type.name}.json"
        val clientAliasJson = JsonAssetReader.readRaw(clientAliasPath, false)

        if (clientJson != null && clientAliasJson != null) {
            data = ClientObfuscationData.parse(clientJson, clientAliasJson)
        }

        if (data != null) {
            cache[type] = data
        }
        return data
    }
}
